src/CMakeLists.txt

```
@@ -173,28 +182,24 @@ endif()
 target_include_directories(${LIBRARY_NAME}
   SYSTEM PUBLIC
   ${EIGEN3_INCLUDE_DIR}
+  ${Boost_INCLUDE_DIRS}
+  ${assimp_INCLUDE_DIR}
   )
```

This one should not be public.
```suggestion
  )
target_include_directories(${LIBRARY_NAME}
  SYSTEM PRIVATE
  ${assimp_INCLUDE_DIR}
  )
```

---

That's right, I did this for the library but I missed it for the include, thanks.

---

I let you take this change the way you like and merge this.